src/jit/codegenxarch.cpp

```diff
@@ -6674,8 +6670,6 @@ void CodeGen::genIntToIntCast(GenTreePtr treeNode)
     }
     else // non-overflow checking cast
     {
-        noway_assert(size < EA_PTRSIZE || srcType == dstType);
```

Comment:
There were two cases how we could come here:
1.	if we came here from widening case, then size was srcSize, for amd64 it could not be greater than 4 for widening. x86 widening to long is the special case that was proceeded in the genLongToIntCast. So, for x86 it was unreachable and didn't check anything useful for amd64.	
2.	If we came here from narrowing, it means that size was dstSize.
a.	If it was cast to smaller type, then dstSize was less srcSize, when srcSize was less or equal EA_PTRSIZE. So, it was true and didn’t check anything useful;
b.	if it was cast with srcSize == dstSize, then it was true for amd64 because we do not perform ulong<->long casts without check. **For x86 there was an error** **because bool -> int cast was treated as narrowing cast (srcSize is size of actual type, that was uint) without check.**